tests/file/test_file_udfs.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import annotations
+
+import io
+from pathlib import Path
+
+import daft
+from daft import DataType as dt
+from daft.functions import file
+
+
+def test_can_convert_string_to_file_type():
+    df = daft.from_pydict({"paths": ["./some_file.txt"]})
+    assert df.schema() == daft.Schema.from_pydict({"paths": dt.string()})
+
+    df = df.select(file(df["paths"]))
+
+    assert df.schema() == daft.Schema.from_pydict({"paths": dt.file()})
+
+
+def test_can_convert_bytes_to_file_type():
+    df = daft.from_pydict({"data": [b"hello world"]})
+    assert df.schema() == daft.Schema.from_pydict({"data": dt.binary()})
+
+    df = df.select(file(df["data"]))
+    assert df.schema() == daft.Schema.from_pydict({"data": dt.file()})
+
+
+def test_can_open_local_file(tmp_path: Path):
+    # Create a file in the temporary directory
+    temp_file = tmp_path / "test_file.txt"
+    temp_file.write_text("test content")
+
+    df = daft.from_pydict({"path": [str(temp_file.absolute())]})
+
+    df = df.select(file(df["path"]))
+
+    @daft.func
+    def read_text(file: daft.File) -> str:
+        with open(file) as f:
+            return f.read()
+
+    df = df.select(read_text(df["path"]).alias("text"))
+    assert df.to_pydict()["text"] == ["test content"]
+
+
+def test_can_open_local_image_with_pil(tmp_path: Path):
+    import numpy as np
+    from PIL import Image
```

Comment:
These, and the other ones, can go into the top level. When they're in the function it causes Python to re-import each thing, which is unnecessary work.